EarablesKIT/EarablesKIT/EarablesKIT/Models/DatabaseService/DatabaseConnection.cs

```diff
@@ -1,39 +1,111 @@
+        /// Constructor of the DatabaseConnection. Initializes the Database and creates the needed table
+        /// </summary>
+        public DatabaseConnection()
         {
-            throw new NotImplementedException();
+            _database = new SQLiteConnection(_path);
+            _database.CreateTable<DBEntryToSave>();
         }
-        
-        public void ExportTrainingsData(string path)
+
+
+
+        /// <inheritdoc />
+        public void DeleteAllEntries()
         {
-            throw new NotImplementedException();
+            _database.DropTable<DBEntryToSave>();
+            _database.CreateTable<DBEntryToSave>();
         }
 
-        public Task<List<DBEntry>> GetAllEntriesAsync()
+
+        /// <inheritdoc />
+        public List<DBEntry> GetAllEntries()
         {
-            throw new NotImplementedException();
+            List<DBEntry> dBEntries = new List<DBEntry>();
+            List<DBEntryToSave> dbEntriesString = _database.Table<DBEntryToSave>().ToList();
+            foreach (var entry in dbEntriesString)
+            {
+                dBEntries.Add(DBEntry.ParseDbEntry(entry));
+            }
+            return dBEntries;
         }
 
-        //TODO Example Documentation
-        /// <summary>
-        /// Method <c>GetMostRecentEntriesAsync</c> connectes to the SQLite Database and gets the amount' recent entries. 
-        /// </summary>
-        /// <param name="amount">The 'amount' lastest database entries</param>
-        /// <returns>A Task countaining a List of database entries wrapped in class <see cref="DBEntry"/></returns>
-        public Task<List<DBEntry>> GetMostRecentEntriesAsync(int amount)
+
+        /// <inheritdoc />
+        public List<DBEntry> GetMostRecentEntries(int amount)
         {
-            throw new NotImplementedException();
+            if (amount <= 0)
+            {
+                return new List<DBEntry>();
+            }
+
+            List<DBEntry> dbEntries = this.GetAllEntries();
+            if (amount >= dbEntries.Count)
+            {
+                return dbEntries;
+            }
+
+            return dbEntries.GetRange(dbEntries.Count-amount, amount);
         }
 
-        public void SaveDBEntry(DBEntry entry)
+        /// <inheritdoc />
+        public int SaveDBEntry(DBEntry entryArg)
+        {
+            int primaryResult;
+            TableQuery<DBEntryToSave> resultQuery = _database.Table<DBEntryToSave>();
+            DBEntry result = null;
+
+            foreach (var dbEntry in resultQuery)
+            {
+                if (dbEntry.DateTime.Equals(entryArg.Date))
+                {
+                    result = DBEntry.ParseDbEntry(dbEntry);
+                    break;
+                }
+            }
+            // DBEntry result = resultQuery.First();
+            if (result == null)
+            {
+                primaryResult = _database.Insert(entryArg.ConvertToDBEntryToSave());
+            }
+            else
+            {
```

Comment:
vielleicht hier noch kurz kommentieren (dass das der Fall ist in dem die Werte einfach addiert werden)